### DISCUSSION TITLE
[V7] how to use matchRoutes in v7?

### DISCUSSION THREAD

**Discussion by jesse996** [remix-run/react-router#12402] [General]
in v6,i have a array of type RouteObject[],and i can use  `matchesRoutes(routes,url)`,but in v7,there is `RouteConfig`， so how to use `matchRoutes` with `RouteConfig`？

**Comment by octet-stream** [comment]
You can try and get these utilities: https://github.com/remix-run/react-router/blob/04a62722fea11baa9ccea682879e530791e28c00/packages/react-router/lib/server-runtime/routes.ts#L19-L56

They're not exposed in public API, so you'll need to copy-paste these to your code.
Then all you need is to call `matchRoutes` with result of `createRoutes` as first parameter:


```ts
const matched = marchRoute(createRoutes(serverBuild.routes), new URL(request.url).pathname, serverBuild.basename)
```

I think I'll open an issue and ask maintainers to add some public utilities to get current route metadata.

Hope this helps.

**Reply by octet-stream** [reply]
Note that `ServerRouteManifest` and `ServerRoute` types are private, but you can easily access them like this:

```ts
import type {ServerBuild} from "react-router"

type ServerRouteManifest = ServerBuild["routes"]

type ServerRoute = NonNullable<ServerRouteManifest[keyof ServerRouteManifest]>
```

**Comment by GabenGar** [comment]
In what context are you trying to match? `Route.ComponentProps["matches"]` already includes pre-matched results.

**Reply by jesse996** [reply]
in some case i want to get the router info like params by url based on my router config file

**Comment by alex-dixon** [comment]
I'm just looking to do this in regular code given route definitions from "routes.ts".

May not be a common ask but having match(routes, path) even just as a gesture to age-old router library formalities would be nice. :)

Copying the above I get this when using route definitions as the argument:
```
TS2345: Argument of type RouteConfigEntry[] is not assignable to parameter of type ServerRouteManifest
Index signature for type string is missing in type RouteConfigEntry[]
```
